### PR TITLE
Add memory cache functions to vmi_instance_t

### LINKED ISSUE #500: Segment Fault in multi vmi_destroy?
I init 2 vmi instance in a process(one thread) which monitoring  processes of VM1 and VM2.
It can get VM1 and VM2 processes correctly.
But when I called "vmi_destroy(vmi_1); " and "vmi_destroy(vim_2); " one by one,  the vmi_1 can be closed all right, the vmi_2 catch a segment fault.

(gdb) bt
#0  0x0000000000000000 in ?? ()
#1  0x00007ffff03b0c26 in memory_cache_entry_free (data=0x7fffcc2080f0) at driver/memory_cache.c:72
#2  0x00007ffff009e8f2 in g_hash_table_remove_all_nodes.part.0 () from /lib64/libglib-2.0.so.0
#3  0x00007ffff009f6fb in g_hash_table_remove_all () from /lib64/libglib-2.0.so.0
#4  0x00007ffff009f73e in g_hash_table_destroy () from /lib64/libglib-2.0.so.0
#5  0x00007ffff03b0fe0 in memory_cache_destroy (vmi=vmi@entry=0x7fffcc150c70) at driver/memory_cache.c:254
#6  0x00007ffff03ab5db in vmi_destroy (vmi=0x7fffcc150c70) at core.c:799

### PR DESCRIPTION
**Pull request by tklengyel**
Fixes #500

### CODE REVIEW

File: libvmi/driver/memory_cache.c [hunk 1]
```diff
@@ -272,8 +265,8 @@ memory_cache_init(
                           size_t),
     unsigned long UNUSED(age_limit))
 {
-    get_data_callback = get_data;
-    release_data_callback = release_data;
+    vmi->get_data_callback = get_data;
```
Comment (smaresca) [inline libvmi/driver/memory_cache.c]: I believe this needs removal of UNUSED around the vmi function parameter

**Comment by cuiyudaniuHT** [comment]
I tried this,  it seems OK. I will try this more.

**Comment by tklengyel** [comment]
I'll be merging this in a week if there are no objections.

**Comment by cuiyudaniuHT** [comment]
I used this commit these days, and it works fine.

**Pull request merged**